### ISSUE TITLE
[Remote-SSH Bug]: Failed to connect to the remote extension host server

### ISSUE THREAD

**Issue by AndreaBellora** [microsoft/vscode-remote-release#8710]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [X] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### Expected Behavior

I would expect to connect to the remote via SSH and be able to work there

### Steps To Reproduce

1. Either in Windows 10 Version 10.0.19044 Build 19044 or Ubuntu 22.04.2 LTS x86_64 on VirtualBox 1.2
2. Open VSCode and install the remote SSH extension
3. Add a new SSH host (some old, already defined, ones seem to work)
4. Try to connect
5. Get the message 'Failed to connect to the remote extension host server (Error: Not an authority that can be resolved!)'

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
[10:18:15.974] Log Level: 2
[10:18:16.004] VS Code version: 1.80.0
[10:18:16.004] Remote-SSH version: remote-ssh@0.102.0
[10:18:16.005] linux x64
```

</p>
</details>

### Anything else?

I get the same behaviour if I try to uninstall the Remote - SSH extension, and I noticed that I cannot see the JSON settings of the Remote extension (but I see the VSCode ones).

**Comment by RDjarbeng** [comment]
@AndreaBellora 
Did you try this fix from #8264. I saw your issue a few hours and went looking for the suggested solutions.

https://github.com/microsoft/vscode-remote-release/issues/8264#issuecomment-1491057876

Like you I was able to connect from the terminal but unable to use the extension to connect.
You didn't specify what remote machine you were using. For me it worked for my raspberry pi 4 and I was able to connect. 

It's been 2 days since I started trying to fix this😒. I hope a permanent fix for this comes out soon.

**Comment by AndreaBellora** [comment]
@RDjarbeng I tried, but it didn't do anything. To me this looks like some communication issue between VSCode and the extension server. It might be that such communication is needed only when setting up new connections (maybe there are some configs to be downloaded?) so that's why it happens only with new connections.

I'm connecting to a remote machine running CentOS Linux 7. It's quite annoying to be fair, but I've been using this extension without issues for more than a year, it's not like I can complain.. I'm using an alternative editor in the meantime.

**Comment by roblourens** [comment]
Sorry for the trouble, this is fixed in the latest Insiders build. https://github.com/microsoft/vscode/issues/187202

Would like to hear more about your use-case though, I'm not sure how you ended up with a URI that contains a `@`

**Comment by AndreaBellora** [comment]
> Would like to hear more about your use-case though, I'm not sure how you ended up with a URI that contains a `@`

@roblourens thank you! 

The reason why I have an `@` in my URI is because I had to define two remotes with different users to connect (and execute code from terminal). The two users have different access rights and functionality on the remote machine, and I need to be able to work remotely as any of them.
Therefore, in the SSH settings, their `HostName` is the same and they have different `User` settings. In order to distinguish them in the Remote Explorer leftbar, I used their `Host` tag (I guess this is what you refer to as URI?) as a label and manually labelled them as `<User>@<HostName>`. I hope this doesn't cause any other troubles under the hood (on a user perspective, it worked fine!)

Is there a recommended way to add labels for better distinction?
